modules/hostap/src/supp_api.c

```diff
@@ -2241,53 +2311,72 @@ int hapd_config_network(struct hostapd_iface *iface,
 	return -1;
 }
 
+static int set_ap_config_params(const struct device *dev, struct wifi_ap_config_params *params)
+{
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_mgmt_api(dev);
+
+	if (!wifi_mgmt_api || !wifi_mgmt_api->ap_config_params) {
+		return -ENOTSUP;
+	}
+
+	return wifi_mgmt_api->ap_config_params(dev, params);
+}
+
 int supplicant_ap_config_params(const struct device *dev, struct wifi_ap_config_params *params)
 {
 	struct hostapd_iface *iface;
-	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_mgmt_api(dev);
 	int ret = 0;
 
-	if (params->type & WIFI_AP_CONFIG_PARAM_MAX_INACTIVITY) {
-		if (!wifi_mgmt_api || !wifi_mgmt_api->ap_config_params) {
-			wpa_printf(MSG_ERROR, "ap_config_params not supported");
-			return -ENOTSUP;
-		}
+	ret = set_ap_config_params(dev, params);
+	if (ret && (ret != -ENOTSUP)) {
+		wpa_printf(MSG_ERROR, "Failed to set ap config params");
+		return -EINVAL;
+	}
 
-		ret = wifi_mgmt_api->ap_config_params(dev, params);
-		if (ret) {
-			wpa_printf(MSG_ERROR,
-				   "Failed to set maximum inactivity duration for stations");
-		} else {
-			wpa_printf(MSG_INFO, "Set maximum inactivity duration for stations: %d (s)",
-				   params->max_inactivity);
-		}
+	k_mutex_lock(&wpa_supplicant_mutex, K_FOREVER);
+
+	iface = get_hostapd_handle(dev);
+	if (!iface) {
```

Comment:
We are trying to migrate away from this style as there is a MISRA rule that recommends to use this style
`iface == NULL` for non boolean variable checks.

Reply:
OK